BDArmory/Weapons/ModuleWeapon.cs

```diff
@@ -4129,7 +4129,11 @@ void CheckAIAutofire()
             }
             if (isAPS)
             {
-                float threatDirectionFactor = Vector3.Dot(fireTransforms[0].position, targetPosition.normalized);
+                float threatDirectionFactor = Vector3.Dot(
```

Comment:
I've added a Vector3 extension for performing the dot product with vectors that need normalising that is around 2x faster (since it only uses a single sqrt and division), so this line would become
```csharp
float threatDirectionFactor = (fireTransforms[0].position - targetPosition).DotNormalized(targetVelocity - part.rb.velocity);
```
Similary for the lines below. Otherwise, it looks good.